SaltyChat.Client/source/Helpers/RoomComparison.ts

```diff
@@ -0,0 +1,36 @@
+import { Player} from "alt-client";
+import { getRoomKeyFromEntity, hasEntityClearLosToEntity } from "natives";
+import { RoomCombinations } from '../Dumps/RoomCombinations'
+
+/**
+ * Compares the roomId of the local player with the roomId of another player and determines the voice compensation factor
+ * @param oPlayer Object of the other player
+ * @returns {} Returns the muffle index of the player, or null if the player is not in a room
+ */
+export function roomComparison(oPlayer: Player) {
+    let roomIdPlayer = getRoomKeyFromEntity(Player.local.scriptID);
+    let roomIdOPlayer = getRoomKeyFromEntity(oPlayer.scriptID);
+    let roomExtension = RoomCombinations.some(roomPairs => {
+        if (Object.values(roomPairs).indexOf(roomIdPlayer) > -1 &&
+            Object.values(roomPairs).indexOf(roomIdOPlayer) > -1) {
+            return true
+        } else { return false;}
+    })
+    
+    // Both players in the same room
+    if (roomIdPlayer == roomIdOPlayer) return null;
```

Comment:
This if should be put directly after the native calls for getRoomKeyFromEntity. Basically if the ids are the same, you're just unneccessarily checking that data